### DISCUSSION TITLE
rocketmq 网络传输协议中magic code 是什么

### DISCUSSION THREAD

**Discussion by dhsjgdf** [apache/rocketmq#8014] [Q&A]
请问网络传输协议中magic code 是什么？我从RemoteCommand源代码中没看到有magic code,我现在要做一个代理，需要判断是否是rocketmq协议，请问怎么来判断

**Comment by leizhiyuan** [comment]
用的是LengthFieldBasedFrameDecoder ，不存在magicCode

看看 encode的过程 org.apache.rocketmq.remoting.protocol.RemotingCommand#encode

**Reply by lizhanhui** [reply]
He is looking for preface of a protocol. For example, SSL/TLS has a negotiation byte: https://tls12.xargs.org/#client-hello/annotated
HTTP2 has connection preface: https://httpwg.org/specs/rfc7540.html#rfc.section.3.5

Unfortunately, RocketMQ lacks this feature and is actually a design flaw.

**Comment by dhsjgdf** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
